backend/algorithms/objects/conditions.py

```diff
@@ -126,9 +135,12 @@ def beneficial(self, user: User, course: dict[str, Tuple[int, int | None]]) -> b
             return False
         return any(c in course.keys() for c in self.courses)
 
-
     def __str__(self) -> str:
-        return f"CoreqCoursesCondition(courses={self.courses}, logic={self.logic})"
+        logic = "and" if self.logic == Logic.AND else "or"
+        return json.dumps({
+            'logic': logic,
+            'children': [f'CoreqCoursesCondition({course})' for course in self.courses],
```

Comment:
could we also use something similar to line 88 here? i.e. ```{ 'id': course }```

Reply:
So children is an array of dictionaries with an id field? Or change children to 'id' which is an array of course code strings.

Reply:
Yeah an array of dictionaries with an id field would be good